Classes/Security/CheckPermissions.php

```diff
@@ -84,6 +90,18 @@ public function checkFileAccess($file, $userFeGroups)
         if ($file->getStorage()->isPublic()) {
             return true;
         }
+
+        /** @var Dispatcher $signalSlotDispatcher */
+        $signalSlotDispatcher = GeneralUtility::makeInstance(Dispatcher::class);
+        $signalSlotDispatcher->dispatch(__CLASS__, 'AddCustomGroups', [&$this]);
```

Comment:
I would prefer a hook or a signal that returns a result. As it is IMO no good practice to call the object from within the signal